WooCommerce/Classes/ViewRelated/Dashboard/Reviews/ReviewsDashboardCardViewModel.swift

```diff
@@ -86,105 +117,162 @@ private extension ReviewsDashboardCardViewModel {
+            return ReviewViewModel(
+                showProductTitle: product != nil,
+                review: review,
+                product: product,
+                notification: notification
+            )
         }
     }
 
-    /// Updates data
     @MainActor
-    func updateReviewsResults() async {
-        let reviews = productReviewsResultsController.fetchedObjects.prefix(Constants.numberOfItems)
-        let productIDs = reviews.map { $0.productID }
+    func synchronizeReviews(filter: ReviewsFilter) async throws {
+        try await synchronizeProductReviews(filter: filter)
+
+        let productIDs = reviews.prefix(Constants.numberOfItems).map { $0.productID }
 
         if productIDs.isNotEmpty {
-            updateProductsResultsControllerPredicate(with: productIDs)
-            do {
-                try await fetchProducts(for: productIDs)
-            } catch {
-                ServiceLocator.crashLogging.logError(error)
-            }
-        }
+            // As long as the app is able to fetch one review once, then the button should always appear.
+            // Later on, when filtering by hold or pending status, the reviews result might become empty.
+            // In that case, the app should keep showing the button to allow the user to see all non-filtered reviews.
+            shouldShowAllReviewsButton = true
 
-        data = reviews
-            .map { review in
-                let product = reviewProducts.first { $0.productID == review.productID }
-                // TODO: also fetch notification
-                return ReviewViewModel(review: review, product: product, notification: nil)
+            // Get product names and, optionally, read status from notifications.
+            await withTaskGroup(of: Void.self) { group in
+                group.addTask { [weak self] in
+                    do {
+                        try await self?.retrieveProducts(for: productIDs)
+                    } catch {
+                        self?.syncingError = error
+                    }
+                }
+                if stores.isAuthenticatedWithoutWPCom == false {
+                    group.addTask { [weak self] in
+                        do {
+                            try await self?.synchronizeNotifications()
+                        } catch {
+                            self?.syncingError = error
+                        }
+                    }
+                }
             }
+        }
     }
 
-    /// Get products from storage if available, if not then fetch remotely.
-    ///
     @MainActor
-    private func fetchProducts(for productIDs: [Int64]) async throws {
-        try productsResultsController.performFetch()
-
-        // Check if all productIDs are available in storage
-        let allProductsAvailable = productIDs.allSatisfy { productID in
-            reviewProducts.contains { $0.productID == productID }
-        }
+    func updateProducts(productIDs: [Int64]) async throws {
```

Comment:
This is not used anymore, do we need to remove it?

Reply:
Reading the code again, I think you want to remove `retrieveProducts` from this call and trigger this method from `updateData()`?

Reply:
Updated this in https://github.com/woocommerce/woocommerce-ios/pull/12769/commits/c3684aeefbc28e9be37bfd67644257ba6879a737

It's supposed to be for calling performFetch() with the latest productID data.